lib/assets/javascripts/cartodb/models/tabledata.js

```diff
@@ -426,26 +417,32 @@ cdb.admin.CartoDBTableData = cdb.ui.common.TableData.extend({
   discreteHistogram: function(nbuckets, column, callback, error) {
 
     var query = 'SELECT DISTINCT(<%= column %>) AS bucket, count(*) AS value FROM (<%= sql %>) _table_sql GROUP BY <%= column %> ORDER BY value DESC LIMIT <%= nbuckets %> + 1';
+    var queryOverMax = 'SELECT count(*) as value FROM (select DISTINCT <%= column %> from <%= table %>) as tmp LIMIT 5000';
 
     var sql = _.template( query, {
       column: column,
       nbuckets: nbuckets,
       sql: this.getSQL(),
     });
-
     this._sqlQuery(sql, function(data) {
 
       var count = data.rows.length;
       var reached_limit = false;
-
       if (count > nbuckets) {
-        data.rows = data.rows.slice(0, nbuckets);
-        reached_limit = true;
+        if(column !== 'the_geom'){
+          reached_limit = true;
+        }
+        var sql = _.template(queryOverMax, {
+          column: column,
+          table: this.table.id
+        });
+        this._sqlQuery(sql, function(data) {
+          callback({ rows: [], reached_limit: reached_limit, total_count: data.rows[0].value });
+        }.bind(this));
+      }else{
```

Comment:
These type of things lint in semi standard (missing spaces around `else`). I'm checking with:

```
npm install -g semistandard-format
semistandard <path to any js file>
```

The editor with easiest installing linter I know of is Atom (package [linter-js-standard](https://atom.io/packages/linter-js-standard))